File: app/src/main/java/com/example/contacts/database/dao/UserDao.java
package com.example.contacts.database.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import com.example.contacts.database.entity.User;

import java.util.List;

@Dao
public interface UserDao {
    @Query("SELECT * FROM user")
    List<User> getAll();

    @Query("SELECT * FROM user WHERE uid IN (:userIds)")
    List<User> loadAllByIds(int[] userIds);

    @Query("SELECT COUNT(*) FROM user WHERE login = :login AND password = :password")
    Integer findCountUsers(String login, String password);

    @Query("SELECT COUNT(*) FROM user WHERE login = :login")
    Integer findUserByLogin(String login);

    @Insert
    void insertAll(User... users);

    @Delete
    void delete(User user);
}

File: app/src/main/java/com/example/contacts/database/dao/UserDao.kt
package com.example.contacts.database.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import com.example.contacts.database.entity.User

@Dao
interface UserDao {
    @get:Query("SELECT * FROM user")
    val all: List<User?>?

    @Query("SELECT * FROM user WHERE uid IN (:userIds)")
    fun loadAllByIds(userIds: IntArray?): List<User?>?

    @Query("SELECT COUNT(*) FROM user WHERE login = :login AND password = :password")
    fun findCountUsers(login: String?, password: String?): Int?

    @Query("SELECT COUNT(*) FROM user WHERE login = :login")
    fun findUserByLogin(login: String?): Int?

    @Insert
    fun insertAll(vararg users: User?)

    @Delete
    fun delete(user: User?)
}